modules/library/main/src/main/java/org/geotools/util/TemporalConverterFactory.java

```diff
@@ -135,8 +135,48 @@ public Object convert(Object source, Class target) throws Exception {
                 };
             }
             
+            if (Long.class.isAssignableFrom(target)) {
```

Comment:
As far as I can tell, the code added here never is true and never gets executed either...